### DISCUSSION TITLE
Guitar Amp Effects Loop

### DISCUSSION THREAD

**Discussion by benjmarshall** [garygru/yummyDSP#8] [Q&A]
Hi Gary,

@garygru 

I've ordered an AI Think Audio Kit to get started playing with this library. I've long wanted to do something in the area of open source guitar effects, but haven't quite found the suitable platform to get started with until now, so thanks for the inspiration!

Quick question as you seem to have experience with this board. It has a headphone output but no specific line level output. I'm hoping to use this board in the effects send/return loop of my guitar amp (selectable level on the amp of -10/+4 dBv). I'm pretty confident a +4dBv line level signal will be fine driving the line level but wanted to ask if you have driven line level output connections from this board? It looks like the AC101 can deliver 18mW max @ 1.8V which I think should be OK into the +4dbV configured effects return. Have you tried something similar?

Cheers,
Ben

**Comment by garygru** [comment]
Hi Ben, 

cool! I've not dived in too much on those boards. Just wanted to see if it runs. 
There are dedicated Line Outputs on the board although not a jack but as some kind of JST / Molex header. 
I would be very surprised if the board can handle +4dBu. So would advise to go for -10dBV. But better check the output to not blow your amp : )

Probably you need to adapt the audioodriver to enable line in and out (_AC101::SetMode(Mode_t mode)?_)
Let me know when you got it running and have fun.
Gary

**Reply by benjmarshall** [reply]
I think the other outputs are speaker outputs aren't they? The doc's [here](https://docs.ai-thinker.com/en/esp32-audio-kit) suggest they are the output of an amplifier IC providing 3W@4Ohm.

**Reply by garygru** [reply]
Sorry you're right. I was misguided by the LOUT label. But yeah doesn't mean line out but left out.
Anyways, perhaps there is a chance to solder a wire from the speaker amp's inputs or AC101 / Module differential outputs.

**Comment by benjmarshall** [comment]
For anyone else watching this thread, or who stumbles across this I have started a discussion over on the Ai Thinker ESP32 Audio Kit repo where I have provided some measurements on safe input ranges:

https://github.com/Ai-Thinker-Open/ESP32-A1S-AudioKit/issues/18
